src/mpi/datatype/typeutil.c

```diff
@@ -28,89 +28,90 @@ static int datatype_attr_finalize_cb(void *dummy);
 typedef struct mpi_names_t {
     MPI_Datatype dtype;
     const char *name;
+    const char *short_name;     /* used in info */
 } mpi_names_t;
-#define type_name_entry(x_) { x_, #x_ }
+#define type_name_entry(x_, shtnm) { x_, #x_, shtnm }
 
 static mpi_names_t mpi_dtypes[] = {
-    type_name_entry(MPI_CHAR),
-    type_name_entry(MPI_UNSIGNED_CHAR),
-    type_name_entry(MPI_SIGNED_CHAR),
-    type_name_entry(MPI_BYTE),
-    type_name_entry(MPI_WCHAR),
-    type_name_entry(MPI_SHORT),
-    type_name_entry(MPI_UNSIGNED_SHORT),
-    type_name_entry(MPI_INT),
-    type_name_entry(MPI_UNSIGNED),
-    type_name_entry(MPI_LONG),
-    type_name_entry(MPI_UNSIGNED_LONG),
-    type_name_entry(MPI_FLOAT),
-    type_name_entry(MPI_DOUBLE),
-    type_name_entry(MPI_LONG_DOUBLE),
-    type_name_entry(MPI_LONG_LONG_INT),
-    type_name_entry(MPI_UNSIGNED_LONG_LONG),
-    type_name_entry(MPI_PACKED),
-    type_name_entry(MPI_LB),
-    type_name_entry(MPI_UB),
-    type_name_entry(MPI_2INT),
+    type_name_entry(MPI_CHAR, "char"),
+    type_name_entry(MPI_UNSIGNED_CHAR, "uchar"),
```

Comment:
I want to keep the type name short and avoid empty space within a name. Otherwise, we have to increase `MPI_MAX_INFO_VAL` (which of course is not a big deal). If we don't care the length, maybe we can even use "MPI_UNSIGNED_CHAR" as the key value. What do you think?